### PR TITLE
Adds test scenarios for Consistency05 and Consisteny06

### PR DESCRIPTION
**Pull request by matsduf**
## Purpose

This PR contains the specification of new test scenarios (five for Consistency05 and two for consistency06) that have been added as unit tests in https://github.com/zonemaster/zonemaster-engine/pull/1340.

## Context

https://github.com/zonemaster/zonemaster-engine/pull/1340

## How to test this PR

Review and test

### CODE REVIEW

File: docs/public/specifications/test-zones/Consistency-TP/consistency05.md [hunk 1]
```diff
@@ -77,16 +82,23 @@ the specific scenario:
 * All responses are authoritative with [RCODE Name] "NoError"
 * EDNS, version 0, is included in all responses on queries with EDNS.
 * EDNS is not included in responses on queries without EDNS.
+* In undelegated data `IPv4` and `IPv6`, repectively, are placeholders for the
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency05.md:85]:
```suggestion
* In undelegated data, `IPv4` and `IPv6`, respectively, are placeholders for the
```

File: docs/public/specifications/test-zones/Consistency-TP/consistency05.md [hunk 2]
```diff
@@ -77,16 +82,23 @@ the specific scenario:
...
+  actual IP addresses used for the scenario. They are to be found where the data
+  is specified.
+  * If no placeholder is given with the name server name, then no IP adress is
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency05.md:88]:
```suggestion
  * If no placeholder is given with the name server name, then no IP address is
```

File: docs/public/specifications/test-zones/Consistency-TP/consistency05.md [hunk 3]
```diff
@@ -127,60 +139,121 @@ Also "happy path". Out-of-bailiwick NS, but with glue.
...
   * "subdomain.addresses-match-7.consistency05.xa" is delegated to the same
     ns1 and ns2.
   * ns1 and ns2 are defined with address records in the "subdomain" zone.
 
+### ADDR-MATCH-DEL-UNDEL-1
+Also the "happy path". But there is an undelegated zone to be tested.
+
+* Zone: addr-match-del-undel-1.consistency05.xa
+  * Delegated zone on ns1 and ns2.
+  * Undelegated zon on ns3 and ns4.
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency05.md:155]:
```suggestion
  * Undelegated zone on ns3 and ns4.
```

File: docs/public/specifications/test-zones/Consistency-TP/consistency05.md [hunk 4]
```diff
@@ -127,60 +139,121 @@ Also "happy path". Out-of-bailiwick NS, but with glue.
...
+  * Delegated zone has neither ns1, ns2, ns3 or ns4 as address records.
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency05.md:156]:
```suggestion
  * Delegated zone has neither ns1, ns2, ns3 nor ns4 as address records.
```

File: docs/public/specifications/test-zones/Consistency-TP/consistency05.md [hunk 5]
```diff
@@ -127,60 +139,121 @@ Also "happy path". Out-of-bailiwick NS, but with glue.
...
+  * Undelegated zone does not have ns1 and ns2 as address records, but it
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency05.md:157]: Do you mean that the undelegated zone can have ns1 or ns2 but not both? Or that it should have neither ns1 nor ns2?

File: docs/public/specifications/test-zones/Consistency-TP/consistency05.md [hunk 6]
```diff
@@ -127,60 +139,121 @@ Also "happy path". Out-of-bailiwick NS, but with glue.
...
+    * ns4.addr-match-del-undel-1.consistency05.xa/IPv4
+    * ns4.addr-match-del-undel-1.consistency05.xa/IPv6
+
+### ADDR-MATCH-DEL-UNDEL-2
+Also the "happy path". But there is an undelegated zone to be tested, and its
+NS are out-of-bailiwick.
+
+* Zone: addr-match-del-undel-2.consistency05.xa
+  * Delegated zone on ns1 and ns2.
+  * Undelegated zon on "ns3.addr-match-del-undel-2.consistency05.xb" and
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency05.md:171]:
```suggestion
  * Undelegated zone on "ns3.addr-match-del-undel-2.consistency05.xb" and
```

File: docs/public/specifications/test-zones/Consistency-TP/consistency05.md [hunk 7]
```diff
@@ -127,60 +139,121 @@ Also "happy path". Out-of-bailiwick NS, but with glue.
...
+    "ns4.addr-match-del-undel-2.consistency05.xb".
+  * Delegated and undelegated zone, respectively, do not have ns1 or ns2 as
+    address records.
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency05.md:174]: Neither ns1 nor ns2, or only exactly one among ns1 or ns2?

File: docs/public/specifications/test-zones/Consistency-TP/consistency05.md [hunk 8]
```diff
@@ -127,60 +139,121 @@ Also "happy path". Out-of-bailiwick NS, but with glue.
...
+    * ns4.addr-match-del-undel-2.consistency05.xb
+
+### ADDR-MATCH-NO-DEL-UNDEL-1
+Also the "happy path". No delegation but there is an undelegated zone to be
+tested.
+
+* Zone: addr-match-no-del-undel-1.consistency05.xa
+  * No delegated zone.
+  * Undelegated zone on ns1 and ns2.
+  * Undelgated data:
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency05.md:186]:
```suggestion
  * Undelegated data:
```

File: docs/public/specifications/test-zones/Consistency-TP/consistency06.md [hunk 9]
```diff
@@ -70,52 +72,85 @@ the specific scenario:
   [RCODE Name] "NoError"
 * EDNS, version 0, is included in all responses on queries with EDNS.
 * EDNS is not included in responses on queries without EDNS.
+* In undelegated data `IPv4` and `IPv6`, repectively, are placeholders for the
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency06.md:75]:
```suggestion
* In undelegated data, `IPv4` and `IPv6`, respectively, are placeholders for the
```

File: docs/public/specifications/test-zones/Consistency-TP/consistency06.md [hunk 10]
```diff
@@ -70,52 +72,85 @@ the specific scenario:
...
+  actual IP addresses used for the scenario. They are to be found where the data
+  is specified.
+  * If no placeholder is given with the name server name, then no IP adress is
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency06.md:78]:
```suggestion
  * If no placeholder is given with the name server name, then no IP address is
```

File: docs/public/specifications/test-zones/Consistency-TP/consistency06.md [hunk 11]
```diff
@@ -70,52 +72,85 @@ the specific scenario:
...
+  * MNAME in SOA on ns2 equal to ns2
+  * Undelegated data:
+    * ns1.mult-soa-mnames-no-del-undel-1.consistency06.xa/IPv4
+    * ns1.mult-soa-mnames-no-del-undel-1.consistency06.xa/IPv6
+    * ns2.mult-soa-mnames-no-del-undel-1.consistency06.xa/IPv4
+    * ns2.mult-soa-mnames-no-del-undel-1.consistency06.xa/IPv6
+
+### MULT-SOA-MNAMES-NO-DEL-UNDEL-2
+Zone not delegated, but there is an undelegated version. Different SOA MNAME on
+the servers. NS are out-of-bailiwing.
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency06.md:139]:
```suggestion
the servers. NS are out-of-bailiwick.
```

File: docs/public/specifications/test-zones/Consistency-TP/consistency06.md [hunk 12]
```diff
@@ -70,52 +72,85 @@ the specific scenario:
...
+
+* Zone: mult-soa-mnames-no-del-undel-2.consistency06.xa.
+  * NS är out-of-bailiwick, "ns3.mult-soa-mnames-no-del-undel-2.consistency06.xb"
```
Comment (marc-vanderwal) [inline docs/public/specifications/test-zones/Consistency-TP/consistency06.md:142]:
```suggestion
  * NS are out-of-bailiwick, "ns3.mult-soa-mnames-no-del-undel-2.consistency06.xb"
```

**Review by marc-vanderwal — COMMENTED** [review]
Looks fine. I just noticed a few typos.

File: docs/public/specifications/test-zones/Consistency-TP/consistency05.md (hunk 5 above)
Comment (matsduf) [reply docs/public/specifications/test-zones/Consistency-TP/consistency05.md:157]: Clarifies.

File: docs/public/specifications/test-zones/Consistency-TP/consistency05.md (hunk 7 above)
Comment (matsduf) [reply docs/public/specifications/test-zones/Consistency-TP/consistency05.md:174]: Clarifies.

**Comment by matsduf** [comment]
@marc-vanderwal and @tgreenx, https://github.com/zonemaster/zonemaster-engine/pull/1340 has been merged and it partly built on this PR. Can this be merged?

**Review by marc-vanderwal — APPROVED** [review]

**Pull request merged**